mesos-client/src/main/scala/com/mesosphere/mesos/conf/MesosClientSettings.scala

```diff
@@ -1,25 +1,57 @@
 package com.mesosphere.mesos.conf
 
-import com.typesafe.config.Config
+import java.net.URL
+
+import com.typesafe.config.{Config, ConfigFactory}
+
+import scala.collection.JavaConverters._
 import scala.concurrent.duration._
 
-case class MesosClientSettings(
-    master: String,
-    redirectRetries: Int = 3,
-    idleTimeout: FiniteDuration = 75.seconds,
-    sourceBufferSize: Int = 10) {}
+class MesosClientSettings private (
+    val masters: Seq[URL],
+    val redirectRetries: Int,
+    val idleTimeout: FiniteDuration,
+    val sourceBufferSize: Int) {
+
+  private def copy(
+      masters: Seq[URL] = this.masters,
+      redirectRetries: Int = this.redirectRetries,
+      idleTimeout: FiniteDuration = this.idleTimeout,
+      sourceBufferSize: Int = this.sourceBufferSize) =
+    new MesosClientSettings(masters, redirectRetries, idleTimeout, sourceBufferSize)
+
+  def withMasters(urls: URL*): MesosClientSettings = copy(masters = urls)
```

Comment:
Let's please just have one method. `withMaster` is unclear whether it appends or replaces the current masters.

Reply:
Java does not support var args. `withMasters(urls: URL*)` is a `withMasters(scala.collection.Seq<URL> urls)` in Java which is annoying to use.